frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FieldSettingsPopover.tsx

```diff
@@ -50,6 +53,14 @@ export function FieldSettingsPopover({
   );
 }
 
+function cleanDefaultValue(fieldType: FieldType, value?: string | number) {
+  if (fieldType === "number") {
+    const clean = Number(value);
+    return !Number.isNaN(clean) ? clean : 0;
```

Comment:
what about this to preserve the ability to have an empty input?

```suggestion
    return !Number.isNaN(clean) && !isEmpty(value) ? clean : '';
```

uses 
```
import { isEmpty } from "metabase/lib/validate";
```

Which is a little construct that I stole from php. don't tell. 🤫

Reply:
if the parameter is optional, shouldn't users be able to leave it blank/null rather than forcing a default value of zero?

Reply:
🤦‍♂️ Right, not enough coffee yet. Thanks!

Reply:
@iethree mind giving this another look please?

Reply:
To be honest, I'm quite confused how we're dealing with required or optional parameters.

### For template tags (SQL questions)
#### Optional 
<img width="361" alt="Screenshot 2023-02-22 at 6 15 31 PM" src="https://user-images.githubusercontent.com/1937582/220719859-e1c0a896-1c83-4b7b-a055-69a814226d5f.png">

#### Required
<img width="355" alt="Screenshot 2023-02-22 at 6 15 39 PM" src="https://user-images.githubusercontent.com/1937582/220719952-788a5e11-5c1f-4f70-90bf-413fc6e41f7e.png">


### For Actions
#### Optional
<img width="404" alt="Screenshot 2023-02-22 at 6 15 21 PM" src="https://user-images.githubusercontent.com/1937582/220720020-d19a2b71-01ad-4c35-8d9a-6ac73f419ab0.png">

#### Required
<img width="419" alt="Screenshot 2023-02-22 at 6 15 16 PM" src="https://user-images.githubusercontent.com/1937582/220720079-c222eb26-c148-4473-9b93-7120493cb3c4.png">


We let users set a parameter default value for template tags when it's **required**, but we do the opposite for Actions–when it's **optional**

The notion of showing a default value when a parameter is **optional** to me feels like we're saying. This input is optional for users, but we need the value regardless, if users don't input anything we'll use the default value here.

And when it's required and we show no default value input it feels like we're saying, this value is required for us so users need to input it nonetheless.

And the notion of showing a default value when the parameter is **required** feels like we're saying this value is required for us, but if users don't input anything use the provided default value.

And when it's optional and we show no default value input it feels like we're saying. This is really optional, we don't need to have the value at all, but if you input it we'll use it.

They're both similar and different in some aspects.

So my question is. Are both behaviors for required/optional parameters for template tags and actions correct at the moment?

Reply:
@iethree I think inherently we all have different ways to look at what these require toggle is for. Yours also makes sense. But want to close the loop here, here's a discussion on our Slack, [product thinks it's a bug](https://metaboat.slack.com/archives/C04FG88HL95/p1677096630128359?thread_ts=1677091265.678819&cid=C04FG88HL95).

Reply:
Thank you both for your reviews @WiNloSt  @iethree 🙏 
I'm going to merge this one and address the required vs. default value relation in another PR

Reply:
Sounds good to me @kulyk